qiskit/circuit/register.py

```diff
@@ -131,7 +131,7 @@ def __getitem__(self, key):
             CircuitError: if the `key` is not an integer.
             QiskitIndexError: if the `key` is not in the range `(0, self.size)`.
         """
-        if not isinstance(key, (int, np.int, np.int32, np.int64, slice, list)):
+        if not isinstance(key, (int, np.int32, np.int64, slice, list)):
```

Comment:
This might be simpler.
```suggestion
        if not isinstance(key, (int, np.integer, slice, list)):
```

Reply:
I am not sure why it does not accept `np.int8` and `np.int16`, if the code owner is ok, please commit it.

Reply:
The original code was committed as part of #4810. `if not isinstance(key, (int, np.integer, slice, list))` looks more appropriate to me. @kdk What do you think?

Reply:
I think this is probably fine, #4810 was a response to the performance regression in #4791 and just selected the types that were in the original issue about numpy types not working. `np.integer` shouldn't have the overhead that `numbers.Integral` introduced and would provide better matching for short types so I think it is a better solution. @t-imamichi do you want to push a follow up PR switching the numpy type checking to this instead?